packages/ui/src/components/organisms/DropdownCardSwitch/DropdownCardSwitch.tsx

```diff
@@ -16,33 +16,67 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, {ReactNode, useState} from 'react';
+import React, {ReactNode, useCallback, useEffect, useState} from 'react';
 import {Dimensions, StyleSheet, View} from 'react-native';
 import {DropdownCard} from '../../molecules';
 
 interface DropdownCardSwitchProps {
   style?: any;
-  dropdownItems: DropdownItem[];
   styleTitle?: any;
+  dropdownItems: DropdownItem[];
+  multiSelection?: boolean;
 }
 interface DropdownItem {
   key: number;
   title: string;
   childrenComp: ReactNode;
+  isDefaultVisible?: boolean;
 }
 
 const DropdownCardSwitch = ({
   style,
-  dropdownItems,
   styleTitle,
+  dropdownItems,
+  multiSelection = false,
 }: DropdownCardSwitchProps) => {
-  const [openedCardKey, setOpenedCardKey] = useState<Number>();
+  const getDefaultVisibleItems = useCallback(() => {
+    let defaultVisibleItems = [];
+
+    if (multiSelection) {
+      defaultVisibleItems = dropdownItems.map(
+        item => item.isDefaultVisible && item.key,
+      );
+    } else {
+      const firstVisibleCard = dropdownItems.find(
+        item => item.isDefaultVisible,
+      );
+      defaultVisibleItems = [firstVisibleCard?.key];
+    }
+
+    return defaultVisibleItems;
+  }, [dropdownItems, multiSelection]);
```

Comment:
Always filter on visible items and then map dropdownItems, and slice array to only get first item if multiSelection mode is not enabled